include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,342 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
+ *                              The effect of changing this value will depend
+ *                              on the key used, the curve used and the
+ *                              underlying implementation, so we provide /c
+ *                              psa_sign_interruptible_get_num_ops() in order
+ *                              to help with tuning this value.
+ *
+ */
+void psa_sign_interruptible_set_max_ops( size_t max_ops );
+
+/**
+ * \brief                       Get the number of basic operations that an
+ *                              operation took to complete.
+ *
+ *                              This is a helper provided to help you tune the
+ *                              value passed to \c
+ *                              psa_sign_interruptible_set_max_ops().
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param ctx                   The \c psa_sign_interruptible_context to use.
+ *                              This must be initialized first.
+ *
+ */
+size_t psa_sign_interruptible_get_num_ops(
+                                          psa_sign_interruptible_context *ctx );
+
+/**
+ * \brief                       Start signing a hash or short message with a
+ *                              private key, in an interruptible manner.
+ *
+ * \see                         \c psa_sign_hash_complete()
+ *
+ * \note                        This function combined with \c
+ *                              psa_sign_hash_complete() is equivalent to
+ *                              \c psa_sign_hash() but it can return early and
+ *                              restart according to the limit set with \c
+ *                              psa_sign_interruptible_set_max_ops() to reduce
+ *                              blocking.
+ *
+ * \note                        Users should call \c psa_sign_hash_complete()
+ *                              on the same context after a successful call to
+ *                              this function until \c psa_sign_hash_complete()
+ *                              either returns 0 or an error. \c
+ *                              psa_sign_hash_complete() will return
+ *                              #PSA_ERROR_IN_PROGRESS if there is more work to
+ *                              do.
+ *
+ * \param[in, out] ctx          The \c psa_sign_interruptible_context to use.
+ *                              This must be initialized first.
+ *
+ * \param key                   Identifier of the key to use
+ *                              for the operation. It must be an asymmetric key
+ *                              pair. The key must allow the usage
+ *                              #PSA_KEY_USAGE_SIGN_HASH.
+ * \param alg                   A signature algorithm (PSA_ALG_XXX
+ *                              value such that #PSA_ALG_IS_SIGN_HASH(\p alg)
```

Comment:
Ok, so I can pass an RSA signature algorithm, for which restartable/interruptible support is not implemented, then `max_ops` will be ignored. I think it's fine, but I'm not sure this is very clear from the documentation. Perhaps the documentation of `set_max_ops` should mention it? Also, what is `get_max_ops` going to return?